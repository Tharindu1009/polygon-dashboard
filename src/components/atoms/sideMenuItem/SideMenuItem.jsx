
import './styles/sideMenuItem.scss';
import * as React from 'react';
import BarChartIcon from '@mui/icons-material/BarChart';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
import PersonOutlineIcon from '@mui/icons-material/PersonOutline';
import SettingsIcon from '@mui/icons-material/Settings';

function SideMenuItem({ title, icon, path, onItemClicked }) {
    return (
        <div data-testid="sideMenuItem" className='sideMenuItem' onClick={() => onItemClicked(path)}>
            {icon === "dashboard" && <BarChartIcon data-testid="dashboardMenuItem" />}
            {icon === "orders" && <ShoppingCartIcon data-testid="ordersMenuItem" />}
            {icon === "account" && <PersonOutlineIcon data-testid="acoountMenuItem" />}
            {icon === "settings" && <SettingsIcon data-testid="settingsMenuItem" />}
            <div className='title'>{title}</div>
        </div>
    );
}

export default SideMenuItem;

