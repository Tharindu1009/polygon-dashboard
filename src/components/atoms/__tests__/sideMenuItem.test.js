import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import SideMenuItem from "../sideMenuItem/SideMenuItem";

afterEach(()=>{
    cleanup();
})
const onClick = jest.fn()

test('should render sideMenuItem component', () => {
    render(<SideMenuItem title="Dashboard" icon="dashboard" path="/home/dashboard" onItemClicked={onClick} />);
    const sideMenuItem = screen.getByTestId('sideMenuItem');
    expect(sideMenuItem).toBeInTheDocument();
    expect(sideMenuItem).toHaveTextContent('Dashboard');
})

test('should render BarChart Icon when passing dashboard as icon prop', () => {
    render(<SideMenuItem title="Dashboard" icon="dashboard" path="/home/dashboard" onItemClicked={onClick} />);
    const dashboardMenuItem = screen.getByTestId('dashboardMenuItem');
    expect(dashboardMenuItem).toBeInTheDocument();
})

test('should render ShoppingCart Icon when passing orders as icon prop', () => {
    render(<SideMenuItem title="Orders" icon="orders" path="/home/orders" onItemClicked={onClick} />);
    const ordersMenuItem = screen.getByTestId('ordersMenuItem');
    expect(ordersMenuItem).toBeInTheDocument();
})

test('should render PersonOutline Icon when passing account as icon prop', () => {
    render(<SideMenuItem title="Account" icon="account" path="/home/account" onItemClicked={onClick} />);
    const acoountMenuItem = screen.getByTestId('acoountMenuItem');
    expect(acoountMenuItem).toBeInTheDocument();
})

test('should render SettingsIcon Icon when passing settings as icon prop', () => {
    render(<SideMenuItem title="Settings" icon="settings" path="/home/settings" onItemClicked={onClick} />);
    const settingsMenuItem = screen.getByTestId('settingsMenuItem');
    expect(settingsMenuItem).toBeInTheDocument();
})

test('should execute callback function when clicking the sideMenuItem', () => {
    render(<SideMenuItem title="Dashboard" icon="dashboard" path="/home/dashboard" onItemClicked={onClick} />);
    const sideMenuItem = screen.getByTestId('sideMenuItem');
    fireEvent.click(sideMenuItem)  
    expect(onClick).toHaveBeenCalledTimes(1);    
})

test('matches sideMenuItem snapshot', () => {
    const tree = TestRenderer.create(<SideMenuItem title="Dashboard" icon="dashboard" path="/home/dashboard" onItemClicked={onClick} />).toJSON();
    expect(tree).toMatchSnapshot();
})