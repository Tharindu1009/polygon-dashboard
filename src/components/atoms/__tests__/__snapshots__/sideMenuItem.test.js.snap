// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches sideMenuItem snapshot 1`] = `
<div
  className="sideMenuItem"
  data-testid="sideMenuItem"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="dashboardMenuItem"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M4 9h4v11H4zm12 4h4v7h-4zm-6-9h4v16h-4z"
    />
  </svg>
  <div
    className="title"
  >
    Dashboard
  </div>
</div>
`;
